students/Cracolici_Jon/lesson09/JonCracolici_OOMailroom.py

```diff
@@ -0,0 +1,326 @@
+    task = input('Please type the number of your selection.')
+    return task
+
+
+def new_donor(new_donor_dict, dB):
+    """Creates a new donor in the database and requests donation ammount."""
+    up_dict = {new_donor_dict['donor_name_key']: {'f_name': new_donor_dict['donor_name_key'].title(),
+                                                  'donor': new_donor_dict['donor'],
+                                                  'total_donated': new_donor_dict['donation'], 'num_donations': 1,
+                                                  'avg_donated': new_donor_dict['donation'],
+                                                  'last_donation': new_donor_dict['donation']}}
+    dB.update(up_dict)
+    return dB
+
+
+def update_donor(new_don_dict, dB):
+    """Updates an existing donor in the database with a new donation."""
+    dB[new_don_dict['donor_name_key']]['total_donated'] += new_don_dict['donation']
+    dB[new_don_dict['donor_name_key']]['num_donations'] += 1
+    dB[new_don_dict['donor_name_key']]['avg_donated'] = dB[new_don_dict['donor_name_key']]['total_donated'] / \
+                                                        dB[new_don_dict['donor_name_key']]['num_donations']
+    dB[new_don_dict['donor_name_key']]['last_donation'] = new_don_dict['donation']
+    return dB
+
+
+def show_current_names(dB):
+    """Displays the names of all previous donors."""
+    names = [item['donor'] for item in dB.values()]
+    # for item in dB.values():
+    #    names.append(item['donor'])
+    last_name = names.pop(len(names) - 1)
+    l = len(names)
+    display = "The current donors are" + (l * " {},").format(*names) + " and {}.".format(last_name)
+    print(display)
+    return  # Returns you to the send a thank you menu
+    # return display #, saty_control(dB)
+
+
+def create_report(dB):
+    """Creates a summary report for all current donors."""
+    dict_list = OrderedDict(sorted(dB.items(), key=lambda x: x[1]['total_donated'], reverse=True))
+    print('Donor Name                | Total Given | Num Gifts | Average Gift')
+    print('------------------------------------------------------------------')
+    for i in dict_list:
+        print('{donor:26}${total_donated:>13.2f}{num_donations:>12} ${avg_donated:>12.2f}'.format(**dB[i]))
+    return
+
+
+def mass_mail(dB):
+    for item in dB.values():
+        print_letters(item)
+    return start_control(dB)
+
+
+def ud_name_handle(a, dB, io='out'):
+    """This function tests whether or not you selected the appropriate database operation by checking to see if
+       the donor is already in the database. Whether or not that is a problem is handled by the io kwarg.
+       args:
+       a = string passed to function by the quit case handler
+       dB = database of donors, a dict of dicts
+       kwargs:
+       io = in or out checker. Defaults to out. Out means that you should not already have the donor.
```

Comment:
I'm unclear what `a` and `io` params represent - recommend refactor / renaming so that the function is more clear.